File: Python/prime.py
#!/usr/bin/env python

def gen (table) :
    tmp = {}
    N = 1000000
    for i in range (0, N) :
        tmp[i] = 1

    for i in range (2, N) :
        if tmp[i]  == 1 :
            for j in range (2, N/i) :
                tmp[i*j] = 0

    for i in range (2, N) :
        if tmp[i] == 1 :
            table.append(i)

    print "done"


if __name__ == '__main__' :
    table = [] 
    gen (table)
    for i in table :
        print "%d, " % i,

File: Perl/work/basic/prime.pl
#!/usr/bin/perl

our @talbe = ();

sub gen {
    @tmp = ();
    $N = 1000000;
    $tmp[$_] = 1 for (0 .. $N);

    for $i (2 .. $N) {
        if ($tmp[$i] == 1) {
            $tmp[$i*$_] = 0 for (2 .. $N/$i);
        }
    }
    for (2 .. $N) {
        push (@table, $_ ) if ($tmp[$_] == 1);
    }
    print "done\n";
}

gen();

print "$_, " for (@table);
